src/fromager/resolver.py

```diff
@@ -234,10 +234,10 @@ def find_matches(self, identifier, requirements, incompatibilities):
             if not self.constraints.is_satisfied_by(identifier, candidate.version):
                 continue
             # Only include sdists if we're asked to
-            if self.include_sdists and not candidate.is_sdist:
+            if self.include_sdists and not candidate.is_sdist and not self.include_wheels:
```

Comment:
I think we want to switch the order of the checks

```suggestion
            if candidate.is_sdist and not self.include_sdists:
```